File: app/public_html/script/functions/samples/22.c
loop(){
    if (getHit()){
        turnToLastHit();
        if (getCloseEnemy())
            attackRanged(getTargetX(), getTargetY());
        else{
            float diff, st = getSimTime();
            do{
                diff = getSimTime() - st;
                stepBack();
            }while(diff < 4 && howManyEnemies() == 0);
        }
    }
    else if (getCloseEnemy())
        attackRanged(getTargetX(), getTargetY());
    else if (moveTo(12.5, 12.5))
        turn(45);
}

File: app/public_html/script/functions/samples/22.py
def loop():
    if getHit():
        turnToLastHit()
        if getCloseEnemy():
            attackRanged(getTargetX(), getTargetY())
        else:
            diff = 0
            st = getSimTime()
            while diff < 4 and howManyEnemies() == 0:
                diff = getSimTime() - st
                stepBack()
     
    elif getCloseEnemy():
        attackRanged(getTargetX(), getTargetY())
    elif moveTo(12.5, 12.5):
        turn(45)
 